core/tx_list.go

```diff
@@ -466,6 +466,43 @@ func (l *txPricedList) Cap(threshold *big.Int, local *accountSet) types.Transact
 	return drop
 }
 
+// FreeSize drops transactions that sum up to at least a certain total size.
+func (l *txPricedList) FreeSize(sizeToFree common.StorageSize, threshold *big.Int, local bool, locals *accountSet) (types.Transactions, common.StorageSize) {
```

Comment:
This method seems a bit leaky. It deletes a bunch of stuff, returns some number, and then the caller will push everything back if the number is inappropriate. Imho we need to make this self contained. If it succeeded in removing enough items, great, return the removed ones. Otherwise if it cannot, it should revert internally. I don't see a reason to return the size removed.

Reply:
I also don't think there's a point in the `local` parameter as it makes things a lot harder to reason about. Instead, the caller can specify `infinity` for the price threshold if the transaction being fitted is a local one. That has the exact same effect of pushing stuff out until there's enough room.